Create public configuration factory methods (#25700)
gradle/gradle: subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/declaring_dependencies.adoc

At this point, we have 3 different configurations with different roles:

> - `someConfiguration` declares the dependencies of my application. It's just a collection of dependencies.

Reviewer: ```suggestion
- `someConfiguration` declares the dependencies of my application. It is simply a collection of dependencies.
```
